internal/transport/transport_test.go

```diff
@@ -2243,3 +2244,27 @@ func TestTCPUserTimeout(t *testing.T) {
 		}
 	}
 }
+
+// TestRequestInfoFoundInStreamContext tests whether data passed in as callheader data gets
+// propagated into the context object as a credentials.RequestInfo object.
+func TestRequestInfoFoundInStreamContext(t *testing.T) {
+	serverConfig := &ServerConfig{}
+	server, client, cancel := setUpWithOptions(t, 0, serverConfig, suspended, ConnectOptions{})
+	defer cancel()
+	defer server.stop()
+	defer client.Close()
+	ch := &CallHdr{
+		Method: "someService/Method",
+	}
+	s, err := client.NewStream(context.Background(), ch)
+	if err != nil {
+		t.Fatalf("client.NewStream() failed: %v", err)
+	}
+	ri, ok := s.ctx.Value(credentials.RequestInfoKey{}).(credentials.RequestInfo)
```

Comment:
Please add a convenience function in `credentials` for the user to do this: `func RequestInfoFromContext(context.Context) RequestInfo` (and unexport `RequestInfoKey`).